rust/index/src/hnsw_provider.rs

```diff
@@ -192,7 +192,7 @@ impl HnswIndexProvider {
         for file in FILES.iter() {
```

Comment:
do we want to fetch the files in parallel as well?
also can remove the todo above 😛

Reply:
woohoo goodbye todo

Reply:
> do we want to fetch the files in parallel as well?
good question. I don't know if the file sizes are homogeneous or one of them is skewingly large and others are tiny. In any case, it should be a different PR